apis/core/lib/middleware/shacl.ts

```diff
@@ -28,22 +34,15 @@ export const shaclMiddleware = (options: ShaclMiddlewareOptions) => asyncMiddlew
       if (pointer.out([sh.targetClass, sh.targetNode, sh.targetObjectsOf, sh.targetSubjectsOf]).values.length === 0) {
         shapes.add($rdf.quad(pointer.term, sh.targetNode, req.hydra.term))
       }
+
+      resource.addOut(rdf.type, pointer.out(sh.targetClass))
```

Comment:
I put this in the loop to ensure that only the root shape's `sh:target` gets added to the resource.